disco_aws_automation/disco_vpc.py

```diff
@@ -601,123 +395,69 @@ def destroy(self):
         self.elasticache.delete_all_cache_clusters(wait=True)
         self.elasticache.delete_all_subnet_groups()
         self._destroy_interfaces()
-        self._destroy_nat_gateways()
+
+        self.disco_vpc_sg_rules.destroy()
+        self.disco_vpc_gateways.destroy_all()
+        DiscoVPCPeerings.delete_peerings(self.get_vpc_id())
         self._destroy_subnets()
-        self._delete_security_group_rules()
-        keep_trying(60, self._destroy_security_groups)
-        self._destroy_igws()
         self._destroy_routes()
-        self._detach_vgws()
-        DiscoVPC.delete_peerings(self.vpc.id)
         return self._destroy_vpc()
 
     def _destroy_instances(self):
         """ Find all instances in vpc and terminate them """
         autoscale = DiscoAutoscale(environment_name=self.environment_name)
         autoscale.clean_groups(force=True)
-
-        instances = [i
-                     for r in self.vpc.connection.get_all_instances(filters=self.vpc_filter())
-                     for i in r.instances]
+        instances = [i['InstanceId']
+                     for r in self.boto3_ec2.describe_instances(Filters=[self.vpc_filter()])['Reservations']
+                     for i in r['Instances']]
 
         if not instances:
             logging.debug("No running instances")
             return
         logging.debug("terminating %s instance(s) %s", len(instances), instances)
 
-        for instance in instances:
-            instance.terminate()
-        for instance in instances:
-            wait_for_state(instance, u'terminated')
+        # for instance in instances:
+        #    instance.terminate()
+        # for instance in instances:
+        #    wait_for_state(instance, u'terminated')
 
+        self.boto3_ec2.terminate_instances(InstanceIds=instances)
+        waiter = self.boto3_ec2.get_waiter('instance_terminated')
+        waiter.wait(InstanceIds=instances,
+                    Filters=[{'Name': 'instance-state-name', 'Values': ['terminated']}])
         autoscale.clean_configs()
 
         logging.debug("waiting for instance shutdown scripts")
-        time.sleep(60)  # see http://copperegg.com/hooking-into-the-aws-shutdown-flow/
```

Comment:
Before: wait for termination + 60 seconds. Now: Wait for termination.

We do loose that 60 seconds wait. I think termination does not wait for graceful instance shutdown.